lib/puppet/resource/type.rb

```diff
@@ -413,6 +413,9 @@ def assign_defaults(resource, scope, resource_hash)
 
   def validate_resource_hash(resource, resource_hash)
     Puppet::Pops::Types::TypeMismatchDescriber.validate_parameters(resource.to_s, parameter_struct, resource_hash)
+  rescue Puppet::ParseError => e
```

Comment:
Agreed, although I think that the "real fix" is part of a bigger refactoring/rewrite that isn't fully scoped and understood yet. Do you want me to add a JIRA suggesting that we use the Puppet Type system for Capabilities?